src/core/ShareDataSet.js

```diff
@@ -144,6 +152,11 @@ ShareDataSet.propTypes = {
    */
   name: PropTypes.string,
 
+  /**
+   * Optional callback function for async loading of input data.
+   */
+  fetchData: PropTypes.func,
+
```

Comment:
I added a callback function for loading async / lazy loading of input data. We could also directly expose `props.data` and add a corresponding `useState()` to update it:
```js
const [inputCollection, setInputCollection] = useState(() => vtkCollection.newInstance());
```

Reply:
It looks like you are just using this to support `<Reader>` + itk-wasm functionality. I think it would be better to have a generic `<TrivialProducer>` component that supports sync/async, leaving ShareDataSet to only focus on sharing datasets.

Reply:
Hmmm makes sense to keep singular responsibility for components. Should the new component continue to be called `<TrivialProducer>` or change to something else like `<Dataset>`. It can either take a `<Reader>` as its child,
OR have `props.fetchData` and `props.data` as optional ways to set input.

Reply:
I like the idea of `<Dataset fetchData={...} OR data={...}>`. `<Reader>` already emits a vtkObject, so it doesn't need to be a child of `<Dataset>`.

Reply:
I have added a new `<Dataset>` for accommodating the direct assignment as well as lazy loading of input data. However, I have left the `vtkTrivialProducer` inside `<ShareDataSet>` as the producers are how it is storing datasets globally.